src/main/java/org/apache/cassandra/sidecar/routes/snapshots/ListSnapshotHandler.java

```diff
@@ -0,0 +1,245 @@
+    {
+        return dataPaths(host, request.keyspace(), request.tableName())
+               .compose(dataDirectoryList ->
+                        builder.streamSnapshotFiles(dataDirectoryList,
+                                                    request.snapshotName(),
+                                                    request.includeSecondaryIndexFiles())
+               )
+               .compose(snapshotFileStream -> buildResponse(host, request, snapshotFileStream));
+    }
+
+    protected Future<List<String>> dataPaths(String host, String keyspace, String table)
+    {
+        return executorPools.service().executeBlocking(() -> {
+            CassandraAdapterDelegate delegate = metadataFetcher.delegate(host);
+            if (delegate == null)
+            {
+                throw cassandraServiceUnavailable();
+            }
+
+            TableOperations tableOperations = delegate.tableOperations();
+            if (tableOperations == null)
+            {
+                throw cassandraServiceUnavailable();
+            }
+
+            return tableOperations.getDataPaths(keyspace, table);
+        });
+    }
+
+    protected Future<ListSnapshotFilesResponse>
+    buildResponse(String host,
+                  SnapshotRequest request,
+                  Stream<SnapshotPathBuilder.SnapshotFile> snapshotFileStream)
+    {
+        int sidecarPort = configuration.port();
+        ListSnapshotFilesResponse response = new ListSnapshotFilesResponse();
+        snapshotFileStream.forEach(file -> {
+            response.addSnapshotFile(new ListSnapshotFilesResponse.FileInfo(file.size,
+                                                                            host,
+                                                                            sidecarPort,
+                                                                            file.dataDirectoryIndex,
+                                                                            request.snapshotName(),
+                                                                            request.keyspace(),
+                                                                            request.tableName(),
+                                                                            file.tableId,
+                                                                            file.name));
+        });
+        return Future.succeededFuture(response);
+    }
+
+    protected Cache<String, Future<ListSnapshotFilesResponse>> initializeCache(CacheConfiguration cacheConfiguration)
+    {
+        if (cacheConfiguration == null)
+        {
+            return null;
+        }
+        return Caffeine.newBuilder()
+                       .maximumSize(cacheConfiguration.maximumSize())
+                       .expireAfterAccess(cacheConfiguration.expireAfterAccessMillis(), TimeUnit.MILLISECONDS)
+                       .recordStats()
```

Comment:
There is no code to collect from `cache.state()`. It will be nice to know the cache hit rate.